sdk/deviceprovisioningservices/azure-iot-deviceprovisioningservice/azure/iot/deviceprovisioningservice/client.py

```diff
@@ -0,0 +1,155 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from typing import TYPE_CHECKING, Any, Union
+
+from azure.core.credentials import AzureNamedKeyCredential, AzureSasCredential
+from azure.core.pipeline import Pipeline
+from azure.core.pipeline.policies import (
+    BearerTokenCredentialPolicy,
+    ContentDecodePolicy,
+    DistributedTracingPolicy,
+    HeadersPolicy,
+    HttpLoggingPolicy,
+    NetworkTraceLoggingPolicy,
+    RetryPolicy,
+    UserAgentPolicy,
+)
+from azure.core.pipeline.transport import (
+    RequestsTransport,
+)  # pylint: disable=no-name-in-module
+from azure.iot.deviceprovisioningservice._generated import (
+    ProvisioningServiceClient as GeneratedProvisioningServiceClient,
+)
+from azure.iot.deviceprovisioningservice._generated._version import VERSION
+
+from ._auth import SasCredentialPolicy, SharedKeyCredentialPolicy
+from ._util import parse_connection_string
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import,ungrouped-imports
+    from azure.core.credentials import TokenCredential
+
+
+class ProvisioningServiceClient(
+    object
+):  # pylint: disable=client-accepts-api-version-keyword
+    """
+    API for connecting to, and conducting operations on a Device Provisioning Service instance
+
+    :param str endpoint: The HTTP endpoint of the Device Provisioning Service instance
+    :param credential: The credential type used to authenticate with the Device Provisioning Service instance
+    :type credential:
+        ~azure.core.credentials.AzureNamedKeyCredential or
+        ~azure.core.credentials.AzureSasCredential or
+        ~azure.core.credentials.TokenCredential
+    """
+
+    def __init__(
+        self,
+        endpoint: str,
+        credential: Union[
+            "TokenCredential", "AzureNamedKeyCredential", "AzureSasCredential"
+        ],
```

Comment:
We need to support an optional api_version keyword-only argument of type string with default value being the latest non-preview API version understood by the client.